osquery/tables/system/windows/users.cpp

```diff
@@ -0,0 +1,125 @@
+  LPUSER_INFO_3 pUserBuffer = nullptr;
+  DWORD dwGenericUserLevel = 3;
+  DWORD dwPreferredMaxLength = MAX_PREFERRED_LENGTH;
+  DWORD dwEntriesRead = 0;
+  DWORD dwTotalEntries = 0;
+  DWORD dwResumeHandle = 0;
+  NET_API_STATUS nEnumStatus;
+
+  nEnumStatus = NetUserEnum(nullptr,
+                            dwGenericUserLevel,
+                            FILTER_NORMAL_ACCOUNT,
+                            (LPBYTE*)&pUserBuffer,
+                            dwPreferredMaxLength,
+                            &dwEntriesRead,
+                            &dwTotalEntries,
+                            &dwResumeHandle);
+
+  // We save the original pointer to the USER_INFO_3 buff for mem management
+  LPUSER_INFO_3 pUserIterationBuffer = pUserBuffer;
+  if (pUserIterationBuffer == nullptr || nEnumStatus != NERR_Success) {
+    if (pUserBuffer != nullptr) {
+      NetApiBufferFree(pUserBuffer);
+    }
+    return results;
+  }
+
+  for (DWORD i = 0; i < dwEntriesRead; i++) {
+    Row r;
+    r["username"] = wstringToString(pUserIterationBuffer->usri3_name);
+    r["description"] = wstringToString(pUserIterationBuffer->usri3_comment);
+    r["uid"] = BIGINT(pUserIterationBuffer->usri3_user_id);
+    r["gid"] = BIGINT(pUserIterationBuffer->usri3_primary_group_id);
+    r["uid_signed"] = r["uid"];
+    r["gid_signed"] = r["gid"];
+    r["shell"] = "C:\\Windows\\system32\\cmd.exe";
+
+    // USER_INFO_23 contains detailed info, like the user Sid
+    DWORD dwDetailedUserLevel = 23;
+    LPUSER_INFO_23 pSidUserBuffer = nullptr;
+    NET_API_STATUS nStatus;
+    nStatus = NetUserGetInfo(nullptr,
+                             pUserIterationBuffer->usri3_name,
+                             dwDetailedUserLevel,
+                             (LPBYTE*)&pSidUserBuffer);
+    if (nStatus != NERR_Success) {
+      if (pSidUserBuffer != nullptr) {
+        NetApiBufferFree(pSidUserBuffer);
+        pSidUserBuffer = nullptr;
+      }
+      continue;
+    }
+
+    LPTSTR sStringSid = nullptr;
+    auto ret =
+        ConvertSidToStringSid(pSidUserBuffer->usri23_user_sid, &sStringSid);
+    if (ret == 0) {
+      if (pSidUserBuffer != nullptr) {
+        NetApiBufferFree(pSidUserBuffer);
+        pSidUserBuffer = nullptr;
+      }
+      continue;
+    }
+    r["uuid"] = sStringSid;
+    std::string query = "SELECT LocalPath FROM Win32_UserProfile where SID=\"" +
+                        std::string(sStringSid) + "\"";
+    WmiRequest wmiRequest(query);
+    std::vector<WmiResultItem>& wmiResults = wmiRequest.results();
+    if (wmiResults.size() != 0) {
+      std::string ret;
+      wmiResults[0].GetString("LocalPath", ret);
+      r["directory"] = ret;
+    }
+    LocalFree(sStringSid);
+    NetApiBufferFree(pSidUserBuffer);
+
+    results.push_back(r);
+    pUserIterationBuffer++;
+  }
+
+  NetApiBufferFree(pUserBuffer);
+
+  if (nEnumStatus == ERROR_MORE_DATA) {
+    LOG(INFO)
+        << "WARNING: NetUserEnum contains mode data. Results may be partial!";
```

Comment:
If you want to log something that only appears with `--verbose` then use:
```
VLOG(1)
```

If you want to log something in a debug build that is not compiled out for a release:
```
DLOG(INFO)
```